src/metax_api/api/rest/base/serializers/catalog_record_serializer.py

```diff
@@ -108,6 +108,7 @@ def is_valid(self, raise_exception=False):
         self.initial_data.pop('next_dataset_version', None)
         self.initial_data.pop('previous_dataset_version', None)
         self.initial_data.pop('removed', None)
+        self.initial_data.pop('date_removed', None)
```

Comment:
Should probably be popped in common_serializer __init__ method instead of here

Reply:
Wouldn't that make date_removed disappear from even GET requests? Do we want that?

Reply:
Should not be the case. Popping stuff from self.initial_data should only affect writes. When reading, data is collected from the ``instance`` variable (see method ``to_representation``).

Reply:
self.initial_data does not seem to be available in init method of common serializer. Anyway, if you take a look at all the other fields in that class's Meta class, they aren't popped either even though the rationale to pop those would be the same as popping the date_removed, i.e. user should not be able to set those values. What happens now if user provides those values?